lib/rubocop/cop/util.rb

```diff
@@ -304,6 +305,14 @@ def to_supported_styles(enforced_style)
           .sub(/^Enforced/, 'Supported')
           .sub('Style', 'Styles')
       end
+
+      private
+
+      def final_pos(src, pos, increment, newlines, whitespace)
+        pos = move_pos(src, pos, increment, true, /[ \t]/)
+        pos = move_pos(src, pos, increment, newlines, /\n/)
+        move_pos(src, pos, increment, whitespace, /\s/)
+      end
```

Comment:
There are no private methods in this Module. Including ones that look pretty scope specific, like `directions` and `move_pos`. Should I start a new trend with this method, or not worry about it and positionally place this closer to where it's invoked?

Reply:
I'd love to clean this file up more. A more desirable interface for `range_with_surrounding_space` seems like it would be an options hash.

```ruby
  # some docs about contents of opts
  def range_with_surrounding_space(range, opts={})
    side ||= opts[:side]
    newlines ||= opts[:newlines]
    whitespace ||= opts[:whitespace]

   etc...
end

def final_pos(src, pos, increment, opts={})
  etc..
end
```

But that would affect any other place `range_with_surrounding_space` is used.  And thus seems like it be better as its own cleanup PR. WDYT?

Reply:
Or keyword arguments. Frankly I don't like option hashes, as the API they expose is not transparent at all. Go figure out which are the supported options. Now that the world has moved on from Ruby 1.9 I think option hashes should die.

Reply:
> There are no private methods in this Module. Some look pretty scope specific, like directions and move_pos. Should I start a new trend with this method, or not worry about it and positionally place this closer to where it's used?
 
Feel free to clean it up. Guess so far nobody just thought about this.

Reply:
Sounds good!

Reply:
> I think option hashes should die.

I second that. And that reminds me to open a PR to make `Style/OptionHash` enabled by default. 😉

Reply:
@garettarrowood: This whole module reeks of utility drawer. I'd rather see stuff removed from it in favour of good abstractions, but we're stuck with it for now.

Reply:
FWIW - Going through the methods, I noticed most of the public methods were really only used once or twice by other Cops. These aren't too tangled into the code's ecosystem.